addons/account/models/account_move_line.py

```diff
@@ -2385,13 +2399,15 @@ def _reconcile_plan_with_sync(self, plan_list, all_amls):
             for results in plan_results:
                 partials_values_list.append(results['partial_values'])
                 if results.get('exchange_values') and results['exchange_values']['move_values']['line_ids']:
-                    exchange_diff_values_list.append(results['exchange_values'])
                     exchange_diff_partial_index.append(partial_index)
                     partial_index += 1
+                    exchange_diff_values_list.append(results['exchange_values'])
 
         # ==== Create the partials ====
         # Link the newly created partials to the plan. There are needed later for caba exchange entries.
         partials = self.env['account.partial.reconcile'].create(partials_values_list)
+        if self._context.get('add_caba_vals'):
+            partials._set_draft_caba_move_vals()
```

Comment:
I don't get why you need to do that here

Reply:
because the caba move can create exchange diff entries as well, and the reconciliation made between the caba move and its exchange diff didn't go through already covered use case (because the caba move isn't identified as such when the reconciliation is made)